src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -94,6 +95,38 @@ public void removePerson(Person key) {
         persons.remove(key);
     }
 
+    /**
+     * Gets person based on their nric
+     * @param nric
+     * @return
+     */
+    public Person getPerson(String nric) {
+        return persons.get(nric);
+    }
+
+    /**
+     * Returns true if a link between the patient and caregiver exists
+     * @param patientNric
+     * @param caregiverNric
+     * @return
+     */
+    public boolean hasLink(String patientNric, String caregiverNric) {
+        Person patient = persons.get(patientNric);
+        Person caregiver = persons.get(caregiverNric);
+        return persons.hasLink(patient, caregiver);
+    }
+
+    /**
+     * Adds a link between the patient and caregiver
+     * @param patientNric
+     * @param caregiverNric
+     */
+    public void addLink(String patientNric, String caregiverNric) {
+        Person patient = persons.get(patientNric);
+        Person caregiver = persons.get(caregiverNric);
```

Comment:
Does persons.get have any way of avoiding a nullpointer reference? I would suggest that you handle this (ideally by using an Optional?)

Reply:
Will do this, thanks for pointing it out